### ISSUE TITLE
utf-8 decode error when parsing some Turbomole files with --multi

### ISSUE THREAD

**Issue by oliver-s-lee** [cclib/cclib#1174]
Some Turbomole output files (for example the output generated by the define program) contain non-utf-8 characters (mostly these are things like the Germanic umlaut character in references). These files will fail to parse when used with the --multi option (the second file is irrelevant, just needed to pass validation):
```
ccget --multi define.out foobar metadata
Attempting to read define.out and dvb
Traceback (most recent call last):
  File "/home/oliver/.local/bin/ccget", line 8, in <module>
    sys.exit(ccget())
  File "/home/oliver/dev/cclib/cclib Src/cclib/scripts/ccget.py", line 178, in ccget
    data = ccread(filename, **kwargs)
  File "/home/oliver/dev/cclib/cclib Src/cclib/io/ccio.py", line 170, in ccread
    return log.parse()
  File "/home/oliver/dev/cclib/cclib Src/cclib/parser/logfileparser.py", line 313, in parse
    for line in inputfile:
  File "/usr/lib/python3.8/fileinput.py", line 248, in __next__
    line = self._readline()
  File "/usr/lib/python3.8/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe4 in position 505: invalid continuation byte
```

But weirdly parses fine when used alone (this might be specific to my machine?):
```
ccget define.out metadata
Attempting to read define.out
metadata
{'basis_set': 'cc-pVDZ',
 'legacy_package_version': '7.5.0',
 'methods': [],
 'package': 'Turbomole',
 'package_version': '7.5.0',
 'success': True}
```

**Comment by oliver-s-lee** [comment]
Eg file: [define.out.zip](https://github.com/cclib/cclib/files/10444340/define.out.zip)

**Comment by berquist** [comment]
Sorry that this slipped through. Would you be able to try fixing it yourself?

**Comment by oliver-s-lee** [comment]
No problem. Yeah I can look into it when I get a sec.

At first look this seems like another `fileinput` problem (considering this is the major difference between single file and multi-file parsing), so a fix might be tricky. Just to get a sense, how married are we to `fileinput`? I've come across some problems with it before (eg, certain crashes during parsing can leave the module in an unusable state and prevent any further reads by the program), so I wonder if maybe this would be a good nudge to look at something else (or look to see if we can use `fileinput` better).

**Comment by berquist** [comment]
I've never heard of `fileinput` until now, but conceptually it sounds simple.  The source of your described problems must come from using `fileinput.input`:
```python
def input(files=None, inplace=False, backup="", *, mode="r", openhook=None,
          encoding=None, errors=None):
    """Return an instance of the FileInput class, which can be iterated.

    The parameters are passed to the constructor of the FileInput class.
    The returned instance, in addition to being an iterator,
    keeps global state for the functions of this module,.
    """
    global _state
    if _state and _state._file:
        raise RuntimeError("input() already active")
    _state = FileInput(files, inplace, backup, mode=mode, openhook=openhook,
                       encoding=encoding, errors=errors)
    return _state
```
when instead we should use an instance of `FileInput` directly, which completely bypasses global state.

**Comment by oliver-s-lee** [comment]
Yeah that makes sense, I'll look into swapping it over

**Comment by berquist** [comment]
- first PR: #1197
- regression: https://github.com/cclib/cclib-data/pull/173

Let's leave this open until https://github.com/cclib/cclib/pull/1197#issuecomment-1526888509 is cleaned up.

**Comment by berquist** [comment]
Closed by #1197
